File: kotlin-code/src/main/kotlin/com/springernature/slides/Slide5.kt
package com.springernature.slides

import java.io.File

fun readFileToString(file: File): String {
    return file.readText()
}

fun doWithText(
    file: File,
    transFun: (String) -> String
): String = transFun(readFileToString(file))


fun main(args: Array<String>) {
    val text1 = doWithText(File("./text.txt")){ txt ->
        txt.uppercase()
    }
    println(text1)

    val text2 = doWithText(File("./text.txt")) {
        it.uppercase()
    }
    println(text2)
}

File: scala-code/src/main/scala/com/springernature/slides/Slide5.scala
package com.springernature.slides

import java.io.{File, FileReader}
import scala.io.Source

object Slide5 {
  def readFileToString(file: File): String =
    Source.fromFile(file).mkString

  def doWithText(
    file: File,
    transFun: String => String
  ): String =
    transFun(readFileToString(file))

  def main(args: Array[String]): Unit = {
    val text1 = doWithText(
      new File("./text.txt"),
      txt => txt.toUpperCase()
    )
    println(text1)

    val text2 = doWithText(
      new File("./text.txt"),
      _.toUpperCase()
    )
    println(text1)
  }
}
